js/RichText/UserTemplatesParameters.js

```diff
@@ -0,0 +1,209 @@
+/**
+ * ---------------------------------------------------------------------
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2015-2021 Teclib' and contributors.
+ *
+ * http://glpi-project.org
+ *
+ * based on GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2003-2014 by the INDEPNET Development Team.
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * GLPI is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * GLPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+/* global tinymce */
+
+var GLPI = GLPI || {};
+GLPI.RichText = GLPI.RichText || {};
+
+/**
+ * User templates parameters autocompleter.
+ *
+ * @since 10.0.0
+ */
+GLPI.RichText.UserTemplatesParameters = class {
```

Comment:
Not sure it is possible, and not sure it would be a good idea for UX, but when a `for` twig tag is deleted maybe all its content should be deleted.

Reply:
Seems complicated, we would need to parse the tinymce input to know where each nodes end.
A possible improvement in the future for sure, but not mandatory in this PR IMO.